### PR TITLE
New fundraising banner and in-app webview donation page

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T333108 & https://phabricator.wikimedia.org/T346823

### Notes
*  This PR updates the UI and actions on our fundraising modal - now the user can choose to be reminded at a later date
* It also keeps the donation flow within the app, opening the pages from an in-app webview
* This PR depends on #4641

### Test Steps
1. Run the experimental app, open it once, background it, and then open it again
2. Set your region to the Netherlands - I was hard-coding "NL" on all methods that verify the country code
3. Add Dutch to your languages
4. Test closing using the "X" button or the "already donated" button (the last one) - it should not show the modal again
5. Delete the app, reinstall, repeat step 1
6. Select maybe later. Check if the reminder toast shows.
7. Visit the article again. it should not show the modal. 
8. Change the date on your phone to tomorrow. Visit the article, it should show the modal.
9. Reinstall and uninstall, repeat step 1. This time, donate.
10. Verify that the `back to article` button shows on the bottom of the Thank you page.
11. Close the web view and verify that the thank you toast shows

### CODE REVIEW

File: Wikipedia/Code/SinglePageWebViewController.swift [hunk 1]
```diff
@@ -170,6 +173,14 @@ class SinglePageWebViewController: ViewController {
...
         navigationController?.popViewController(animated: true)
     }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        if didReachThankyouPage {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+                WMFAlertManager.sharedInstance.showBottomAlertWithMessage(CommonStrings.donateThankTitle, subtitle: CommonStrings.donateThankSubtitle, image: UIImage.init(systemName: "heart.fill"), type: .custom, customTypeName: "donate-success", duration: -1, dismissPreviousAlerts: true)
+            }
+        }
+    }
```
Comment (mazevedofs) [inline Wikipedia/Code/SinglePageWebViewController.swift:183]: This feels a little hacky, so I'm open to suggestions here :D

**Review by tonisevener — CHANGES_REQUESTED** [review]
Thanks! Looking great. This first round is a code-only review, I'll test it tomorrow and will let you know if I find anything else. Mainly just small naming tweaks and moving bits of code around.

File: Wikipedia/Code/SinglePageWebViewController.swift [hunk 2]
```diff
@@ -52,7 +54,25 @@ class SinglePageWebViewController: ViewController {
...
+        let contentView = UIView(frame: .zero)
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        contentView.backgroundColor = self.theme.colors.paperBackground
+        return contentView
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setTitle(CommonStrings.returnToArticle, for: .normal)
```
[the remark this answers is not in the document]
Comment (mazevedofs) [reply Wikipedia/Code/SinglePageWebViewController.swift:68]: I'll leave a TODO on this one for now, and I can do a fix PR after yours is up

File: Wikipedia/Code/ArticleViewController+Announcements.swift [hunk 3]
```diff
@@ -78,41 +78,37 @@ extension ArticleViewController {
...
-        alert.addAction(maybeLaterAction)
-        alert.addAction(alreadyDonatedAction)
-        alert.addAction(closeAction)
-        
-        present(alert, animated: true)
+
+        wmf_showFundraisingAnnouncement(theme: theme, asset: asset, primaryButtonTapHandler: { sender in
+            if let url = asset.actions[0].url {
+                self.navigate(to: url, useSafari: false)
+            }
```
[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/ArticleViewController+Announcements.swift:89]: @mazevedofs Let me know if you have trouble displaying this alert view. There are quite a few hoops to jump through to get to a state where the app thinks a user qualifies for this prompt. Going through the Test steps "Prep work:" and then the "testing locally" steps 1-3 on the https://github.com/wikimedia/wikipedia-ios/pull/4641 description should get things set up for displaying this prompt.

[the remark this answers is not in the document]
Comment (tonisevener) [reply Wikipedia/Code/ArticleViewController+Announcements.swift]: @mazevedofs FYI, I noticed a bug with this presentation logic on my code. I pushed a fix just now, hope that's okay!

**Review by tonisevener — APPROVED** [review]

**Pull request merged**